std/experimental/checkedint.d

```diff
@@ -1643,17 +1643,32 @@ static:
 {
     auto a = checked!Warn(int.min);
     auto b = checked!Warn(-1);
-    assert(a / b == a * b);
-}
-
-@system unittest
-{
-    import std.exception : assertThrown;
-    import core.exception : AssertError;
+    auto x = checked!Abort(int.min);
+    auto y = checked!Abort(-1);
 
-    auto a = checked!Abort(int.min);
-    auto b = checked!Abort(-1);
-    assertThrown!AssertError(a / b);
+    // Temporarily redirect output to stderr to make sure we get the right output.
+    import std.process : uniqueTempPath;
+    import std.stdio : stderr;
+    auto tmpname = uniqueTempPath;
+    auto t = stderr;
+    stderr.open(tmpname, "w");
+    // Open a new scope to minimize code ran with stderr redirected.
+    {
+        scope(exit) stderr = t;
+        assert(a / b == a * b);
+        import std.exception : assertThrown;
+        import core.exception : AssertError;
+        assertThrown!AssertError(x / y);
+    }
+    import std.file : readText;
+    auto witness = readText(tmpname);
+    auto expected =
+"Overflow on binary operator: int(-2147483648) / const(int)(-1)\r\n" ~
+"Overflow on binary operator: int(-2147483648) * const(int)(-1)\r\n" ~
+"Overflow on binary operator: int(-2147483648) / const(int)(-1)\r\n";
+    import std.string : replace;
+    version(Posix) expected = expected.replace("\n\r", "\n");
```

Comment:
Why not use `newline` from std.ascii directly (above)?

Reply:
noice